### PR TITLE
[5247] Data migration to fix disability information for Apply trainees for 2021/2022

### PR DESCRIPTION
**Pull request by jhenderson**
### Context
https://trello.com/c/2Kdlvra1/5247-m-apply-to-disability-21-22-backfill

### Changes proposed in this pull request
- Data migration file
- Uploaded CSV file to production so it can be processed

### Guidance to review
- Tested it locally using a local CSV file. Works as expected.

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by jhenderson** [comment]
[[M] Apply to Disability 21/22 Backfill](https://trello.com/c/2Kdlvra1/5247-m-apply-to-disability-21-22-backfill)

File: db/data/20230214121925_backfill_21_to_22_disabilities_from_apply.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          trainee.trainee_disabilities.each_with_index do |trainee_disability, index|
+            if trainee_disability.disability.name == Diversities::OTHER
+              trainee_disability.update(additional_disability: JSON(row[:disabilities])[index]&.strip)
+            end
+          end
+        else
+          trainee.disability_disclosure = Diversities::DISABILITY_DISCLOSURE_ENUMS[:no_disability]
+          trainee.disabilities = []
+          trainee.save
+        end
```
Comment (AbigailMcP) [inline db/data/20230214121925_backfill_21_to_22_disabilities_from_apply.rb:34]: do we also need to make sure `diversity_disclosure` is set to `diversity_disclosed` in both cases?

File: db/data/20230214121925_backfill_21_to_22_disabilities_from_apply.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class Backfill21To22DisabilitiesFromApply < ActiveRecord::Migration[7.0]
+  def up
+    upload = Upload.find_by(id: 6)
```
Comment (AbigailMcP) [inline db/data/20230214121925_backfill_21_to_22_disabilities_from_apply.rb:5]: are there any uploads in other deployed envs? maybe making sure this runs only in prod would be good. not sure whether it matters too much though!

File: db/data/20230214121925_backfill_21_to_22_disabilities_from_apply.rb (hunk 1 above)
Comment (jhenderson) [reply db/data/20230214121925_backfill_21_to_22_disabilities_from_apply.rb:34]: Good spot! 👍

File: db/data/20230214121925_backfill_21_to_22_disabilities_from_apply.rb (hunk 2 above)
Comment (jhenderson) [reply db/data/20230214121925_backfill_21_to_22_disabilities_from_apply.rb:5]: `production` and `productiondata` are the same and there's no uploads at all in the `staging` environment so I we're good.

**Review by AbigailMcP — APPROVED** [review]

**Pull request merged**